hvplot/plotting/core.py

```
@@ -822,6 +822,8 @@ def heatmap(self, x=None, y=None, C=None, colorbar=True, **kwds):
             Whether to apply log scaling to the z-axis. Default is False.
         reduce_function : function, optional
             Function to compute statistics for heatmap, for example `np.mean`.
+            If omitted, no aggregation is applied and duplicate values are dropped.
+            Note: Explicitly setting this parameter to `None` is not allowed and will raise an error.
```

Hmm this sounds like a bug :) Or put differently, why should hvPlot raise an error if this option is set to `None`? But I think I understand now why you didn't add it to the signature. Is it because it would raise this error?

---

> Hmm this sounds like a bug :) Or put differently, why should hvPlot raise an error if this option is set to `None`? But I think I understand now why you didn't add it to the signature. Is it because it would raise this error?

Yes, although I thought it was by design and not a bug.

---

I think it's more about how hvPlot is implemented rather than an API design choice in this case. Options that don't have an explicit default (e.g. set in the signature, obtained with `self.kwds.get(<option>, DEFAULT)`) aren't necessarily treated consistently.

```python
        if 'reduce_function' in self.kwds:
            hmap = hmap.aggregate(function=self.kwds['reduce_function'])
```

`reduce_function` appears to be enabled when set to whatever value. It is interesting to get an error when passing down `function=None` as the signature of `Dataset.aggregate` seems to allow that (might be a bug in HoloViews).

```python
    def aggregate(self, dimensions=None, function=None, spreadfn=None, **kwargs):
```

Ignoring this error, setting `reduce_function=None` in hvPlot should likely be a no-op.

```python
        if self.kwds.get('reduce_function', None) is not None:
            hmap = hmap.aggregate(function=self.kwds['reduce_function'])
```

Let's not do that in this PR, we can separately work in improving how hvPlot deals with all these options and their default values.